### ISSUE TITLE
Using pretrained Bert Model in CPU Only Environment - How to speed up Predictions?

### ISSUE THREAD

**Issue by Stamtiniakos** [amaiya/ktrain#267]
Not  a bug, I am posting this question here in case that someone has faced the same problem before and knows the answer.

I have trained a ktrain Bert text classification model using GPU on Azure.

Example of the code can be found here: `https://github.com/lukasgarbas/nlp-text-emotion/blob/master/bert.ipynb`

I saved the model into files (.h5 and .preproc) for later use. The files are big e.g. 1.2 gb .

I loaded the model in a computer where only CPU is available. The model works fine but the `model.predict(text)` function is super slow predicting the class of the text e.g. on average 1 tweet sized message per second.

Adding GPU on the computer is not an option. I wonder if there is another way to make it run faster? e.g. train the models in a different way (without compromising accuracy) or save the model in a different file format?

**Comment by amaiya** [comment]
BERT is a very large model that will always be slower on a CPU.    You can try using `distilbert` which has half the parameters but nearly the same performance.  Or, you can try a smaller non-transformers model like NBSVM that exhibits extremely fast training and inference on a CPU and achieves 92% accuracy on IMDB dataset (vs. 94% accuracy of BERT).   The `load_predictor` function has a `batch_size` parameter that is used when making predictions, so you can also try playing around with that, although not sure how much impact it'll have on CPU.  Trying quantization and other things are also options.  See [this blog post](https://blog.roblox.com/2020/05/scaled-bert-serve-1-billion-daily-requests-cpus/).
